gateway/1.8/modules/ROOT/pages/_partials/task-config-readiness-liveness.adoc

```diff
@@ -0,0 +1,191 @@
++
+The default Kubernetes readiness probe is configured to be error-tolerant. If a configuration error occurs (for example, if you misconfigure a policy), the readiness state still changes to `true` once the API instances and policies are deployed. This enables Flex Gateway to scale even though there might be configuration errors.
++
+You can also configure the readiness probe to be error-intolerant, meaning that misconfigured gateway instances never reach a readiness state of `true`.
++
+To configure the Helm chart for error intolerance, remove the following flags from the readiness probe: `--allow-api-errors`, `--allow-envoy-errors`, and `--allow-policy-errors`. In deployments that query the readiness state through HTTP requests, use query parameters: `allowEnvoyErrors=false`, `allowAPIErrors=false`, and `allowPolicyErrors=false`.
+
+Docker, Podman, and Linux::
++
+For Docker and Linux deployments, Flex Gateway does not include a pre-configured default readiness probe. However, you can still run the readiness probe command manually, or configure it for automated use with third-party services. Besides the Kubernetes readiness probe, MuleSoft does not provide support for third-party readiness probes.
++
+To expose the readiness API to external components, apply the following configuration to the Flex Gateway instance:
++
+[source,yaml]
+----
+apiVersion: gateway.mulesoft.com/v1alpha1
+kind: Configuration
+metadata:
+ name: probe
+spec:
+ probe:
+   enabled: true
+   port: 3000
+----
+====
+
+Readiness is also probed during shutdown, at which point a `Connection: close` header is sent. A `SIGTERM` signal initiates a drain of downstream connections by signaling them to reconnect. The default drain period is 25 seconds, after which Flex Gateway exits gracefully.
+
+You can modify a drain period via the `FLEX_SERVICE_ENVOY_DRAIN_TIME` environment variable in Docker and Linux, or the `gateway.drainSeconds` option in a Kubernetes Helm chart.
+
+Flex Gateway is preconfigured for a graceful shutdown of 30 seconds, which is 5 seconds more than the drain period. When increasing the drain period, also increase the shutdown period using `TimeoutStopSec` in Linux or `terminationGracePeriodSeconds` in Kubernetes. The shutdown period must exceed the drain period.
+
+The default readiness probe in Kubernetes runs every 10 seconds with a failure threshold of 2, allowing shutdown detection within 20 seconds. This ensures that no new traffic is sent to the instance shutting down. Similar configurations must be used with readiness probes in other environments like AWS Load Balancer.
+
+[[configure-a-liveness-probe]]
+== Configure a Liveness Probe
+
+To test whether a Flex Gateway instance is operational, you can use the following CLI command:
+
+[source,ssh]
+----
+flexctl probe --check=liveness
+----
+
+If the Flex Gateway instance is operational, the command returns an exit code of `0`. Otherwise the command returns an exit code of `1`.
```

Comment:
```suggestion
If the Flex Gateway instance is operational, the command returns an exit code of `0`. Otherwise, the command returns an exit code of `1`.
```